stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/app.py

```diff
@@ -24,6 +24,11 @@
 settings = ElasticsearchSettings()
 session = Session.create_from_settings(settings)
 
+filter_extension = FilterExtension(client=EsAsyncBaseFiltersClient())
+filter_extension.conformance_classes.append(
+    "http://www.opengis.net/spec/cql2/1.0/req/advanced-comparison-operators"
```

Comment:
this shouldn't be advertised unless `IN` and `BETWEEN` are also implemented.

Reply:
Okay, I have implemented `IN` and `BETWEEN` in the latest commit. 

The conformance class endpoint above aligns with what [STAC Browser expects](https://github.com/radiantearth/stac-browser/blob/main/src/components/ApiCapabilitiesMixin.js#L12C1-L12C103). The [Filter extension README](https://github.com/stac-api-extensions/filter?tab=readme-ov-file#conformance-classes) has the conformance class endpoint as `http://www.opengis.net/spec/cql2/1.0/conf/advanced-comparison-operators`. Which should be used? I get a 404 response from both...


I followed [this example](https://github.com/radiantearth/stac-api-spec/tree/v1.0.0-beta.5/fragments/filter#example-10-using-between) which has the `BETWEEN` bounds being passed as an array. 

```json
{
  "filter-lang": "cql2-json",
  "filter": {
    "op": "between",
    "args": [
      { "property": "eo:cloud_cover" },
      [ 0, 50 ]
    ]
  }
}
```

This required adjustments to the `Clause` filter class to allow lists and some typing checks in `IN` and `BETWEEN`

```python
class Clause(BaseModel):
    """Filter extension clause."""

    op: Union[LogicalOp, ComparisonOp, AdvancedComparisonOp, SpatialIntersectsOp]
    args: List[Union[Arg, List[Arg]]]
```

```python
elif self.op == AdvancedComparisonOp.between:
            if not isinstance(self.args[1], List):
                raise RuntimeError(f"Arg {self.args[1]} is not a list")
            return {
                "range": {
                    to_es(self.args[0]): {
                        "gte": to_es(self.args[1][0]),
                        "lte": to_es(self.args[1][1]),
                    }
                }
            }
elif self.op == AdvancedComparisonOp._in:
    if not isinstance(self.args[1], List):
        raise RuntimeError(f"Arg {self.args[1]} is not a list")
    return {
        "terms": {to_es(self.args[0]): [to_es(arg) for arg in self.args[1]]}
    }
```

Reply:
> Okay, I have implemented `IN` and `BETWEEN` in the latest commit.

nice, will review.

> The conformance class endpoint above aligns with what [STAC Browser expects](https://github.com/radiantearth/stac-browser/blob/main/src/components/ApiCapabilitiesMixin.js#L12C1-L12C103). The [Filter extension README](https://github.com/stac-api-extensions/filter?tab=readme-ov-file#conformance-classes) has the conformance class endpoint as `http://www.opengis.net/spec/cql2/1.0/conf/advanced-comparison-operators`. Which should be used? I get a 404 response from both...

Ah, that seems to be a bug in STAC Browser - they should be the urls with `conf` in them, not `req`. They aren't resolvable as a URL -- many of the OGC URL-like conformance classes aren't. 

> I followed [this example](https://github.com/radiantearth/stac-api-spec/tree/v1.0.0-beta.5/fragments/filter#example-10-using-between) which has the `BETWEEN` bounds being passed as an array.

This has been changed in newer versions of CQL2 (or v1.0.0-beta.5 had bad examples?). The current version of Filter Extension is here, specifically BETWEEN: https://github.com/stac-api-extensions/filter/blob/main/README.md#example-11-using-between
I think you'll probably need to make some changes to work with this instead of the beta.5 version.